FWCore/Framework/src/EventProcessor.cc

```diff
@@ -2090,40 +1973,21 @@ namespace edm {
   }
 
   void EventProcessor::writeProcessBlockAsync(WaitingTaskHolder task, ProcessBlockType processBlockType) {
-    using namespace edm::waiting_task;
-    chain::first([&](auto nextTask) {
-      ServiceRegistry::Operate op(serviceToken_);
-      schedule_->writeProcessBlockAsync(
-          nextTask, principalCache_.processBlockPrincipal(processBlockType), &processContext_, actReg_.get());
-    }) | chain::ifThen(not subProcesses_.empty(), [this, processBlockType](auto nextTask) {
-      ServiceRegistry::Operate op(serviceToken_);
-      for (auto& s : subProcesses_) {
-        s.writeProcessBlockAsync(nextTask, processBlockType);
-      }
-    }) | chain::runLast(std::move(task));
+    ServiceRegistry::Operate op(serviceToken_);
+    schedule_->writeProcessBlockAsync(
+        task, principalCache_.processBlockPrincipal(processBlockType), &processContext_, actReg_.get());
```

Comment:
The `task` could be moved
```suggestion
        std::move(task), principalCache_.processBlockPrincipal(processBlockType), &processContext_, actReg_.get());
```

Reply:
I think we need to not move so that the ServiceRegistry::Operate goes out of scope before the task goes out of scope. I believe there is the extremely unlikely possibility the Services stay on a very long time and cause problems during shutdown, because once we call doneWaiting everything else continues to completion and we are off to the races. IIRC, we actually saw that problem in the past in some rare cases.

Reply:
Ok. Could you then add a comment here that the lifetime of the `task` must be longer than the lifetime of `op` for that reason?

Reply:
There 10 or 20 places in the code like this in EventProcessor. Should I mark them all? Maybe a good idea. This is something easy to forget about.

Reply:
I added the two comments you suggested. If there is more related to this I would suggest deferring it until I submit the PR dealing with SubProcess and Services. Let me know about if there is something more you want me to do and I'll get it into the first version of that PR. (That is assuming it isn't something that is going to cause an immediate problem.)